netrd/reconstruction/partial_correlation_influence.py

```diff
@@ -0,0 +1,187 @@
+"""
+partial_correlation_influence.py
+---------------------
+
+Reconstruction of graphs using the partial correlation influence, as defined in:
+
+Kenett, D. Y. et al. Dominating clasp of the financial sector revealed by
+partial correlation analysis of the stock market. PLoS ONE 5, e15032 (2010).
+
+The index variable option as in:
+
+Kenett, D. Y., Huang, X., Vodenska, I., Havlin, S. & Stanley, H. E. Partial correlation
+analysis: applications for financial markets. Quantitative Finance 15, 569–578 (2015).
+
+
+author: Carolina Mattsson
+email: mattsson dot c at northeastern dot edu
+Submitted as part of the 2019 NetSI Collabathon
+"""
+from .base import BaseReconstructor
+import numpy as np
+import networkx as nx
+from scipy import stats, linalg
+from ..utilities import create_graph, threshold
+
+
+class PartialCorrelationInfluenceReconstructor(BaseReconstructor):
+    def fit(self,
+            TS,
+            index=None,
+            threshold_type='range',
+            **kwargs):
+        """
+        Reconstruct a network from time series data using the
+        *average effect of a series Z on the correlation between
+        a series X and all other series*
+
+        The partial correlation influence:
+        d(X:Z)=<d(X,Y:Z)>_Y!=X where
+        d(X,Y:Z) = ρ(X,Y) - ρ(X,Y:Z)
+
+        If an index is given, both terms become partial correlations:
+        d(X,Y:Z) ≡ ρ(X,Y:M) − ρ(X,Y:M,Z)
+
+        Params
+        ------
+
+        index (int, array of ints, or None): Take the partial correlations of
+        each pair of elements holding constant an index variable or set of
+        index variables. If None, take the partial correlations of the
+        variables holding constant all other variables.
+
+        threshold_type (str): Which thresholding function to use on the matrix of
+        weights. See `netrd.utilities.threshold.py` for documentation. Pass additional
+        arguments to the thresholder using `**kwargs`.
+
+        Returns
+        -------
+        G: a reconstructed graph.
+
+        """
+
+        if index:
+            p_cor = partial_corr(TS, index=index)
+            n_TS  = p_cor.shape[0]
+            p_cor = np.delete(p_cor, index, axis=0)
+            p_cor = np.delete(p_cor, index, axis=1)
+        else:
+            p_cor = partial_corr(TS)
+
+        p_cor = np.fill_diagonal(p_cor, float("nan"))
```

Comment:
This is the problematic line. `np.fill_diagonal` modifies an array in-place and returns `None`, so the fix should be to change the line to simply `np.fill_diagonal(p_cor, float('nan'))`.

Reply:
Thanks! That's kind of inconsistent of numpy, if I'm allowed to say so. 

I think that does it? My local tests say PASSED all the way down, but I also can't really parse the warnings it's giving. Is there a sample TS object that I can just load and feed through the code line by line in a terminal window? But maybe it's whatever - it passes and we'll notice?